core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -572,6 +574,8 @@ object KafkaConfig {
   val MaxConnectionsPerIpDoc = "The maximum number of connections we allow from each ip address. This can be set to 0 if there are overrides " +
     "configured using " + MaxConnectionsPerIpOverridesProp + " property"
   val MaxConnectionsPerIpOverridesDoc = "A comma-separated list of per-ip or hostname overrides to the default maximum number of connections. An example value is \"hostName:100,127.0.0.1:200\""
+  val MaxConnectionsDoc = "The maximum number of connections we allow for each listener at any time. This limit is applied in addition to any per-ip " +
+    "limits configured using " + MaxConnectionsPerIpProp + ". The config name may be prefixed with the listener prefix to specify different limits for different listeners."
```

Comment:
Maybe add an explanation of how max.connections is enforced vs how listener.max.connections, and how we expect max.connections to be relatively high and set based on broker capacity, while listener.max.connections is to allocate connections to applications and should be set based on application requirements.

It took me few extra explanations to get it, so I think a bit more of explanation here will help.

Reply:
@gwenshap Thanks for the review, I have updated the doc.